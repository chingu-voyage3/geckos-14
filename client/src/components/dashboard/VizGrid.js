/** VizGrid Component :
The VizGrid componet renders for now a basic CSS grid
TODO: Transform grid into draggable
 */
import React from 'react';
import Viz from './Viz';
// css files included in RGL package
import '../../assets/react-grid-layout.css';
import '../../assets/react-resizable.css';
import * as d from '../../tempData.js';

import ReactGridLayout from 'react-grid-layout';

const renderViz = viz => {
  // console.log('renderViz', viz);
  return <Viz key={viz.name} {...viz} />;
};
const renderVizGrid = vizs => {
  // console.log('renderVizList');
  return vizs.map(viz => {
    return renderViz(viz);
  });
};
const VizGrid = props => {
  // console.log('VizGrid Component');
  return (
    <div className="">
      <ReactGridLayout className="layout" layout={d.gridLayout} autoSize={true} width={900}>
        {renderVizGrid(props.vizs)}
      </ReactGridLayout>
    </div>
  );
};

export default VizGrid;
