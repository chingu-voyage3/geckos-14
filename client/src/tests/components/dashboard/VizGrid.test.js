import React from 'react';
import VizGrid from '../../../components/dashboard/VizGrid';

import { shallow } from 'enzyme';
const lineData = [
  { date: '12/12', temp: 2 },
  { date: '13/12', temp: 0 },
  { date: '14/12', temp: 5 },
  { date: '15/12', temp: -3 }
];
const barData = [
  { date: '12/12', hum: 95 },
  { date: '13/12', hum: 93 },
  { date: '14/12', hum: 98 },
  { date: '15/12', hum: 85 }
];
const vizs = [
  {
    id: 'viz0',
    name: 'Temp Line',
    deviceName: 'LondonTemp',
    model: 'VictoryLine',
    x: 'date',
    y: 'temp',
    data: lineData
  },
  {
    id: 'viz1',
    name: 'Humidity Bar',
    deviceName: 'LondonHumidity',
    model: 'VictoryBar',
    x: 'date',
    y: 'hum',
    data: barData
  }
];
const update = jest.fn();
it('renders without crashing', () => {
  const wrapper = shallow(<VizGrid vizs={vizs} actions={{ update: update }} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
