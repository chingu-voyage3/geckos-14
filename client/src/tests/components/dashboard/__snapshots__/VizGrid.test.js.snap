// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className=""
>
  <ReactGridLayout
    autoSize={true}
    className="layout"
    cols={3}
    compactType="vertical"
    containerPadding={null}
    draggableCancel=""
    draggableHandle=""
    isDraggable={true}
    isResizable={true}
    layout={
      Array [
        Object {
          "h": 1,
          "i": "Temp Line",
          "w": 1,
          "x": 0,
          "y": 0,
        },
        Object {
          "h": 1,
          "i": "Humidity Bar",
          "w": 1,
          "x": 1,
          "y": 0,
        },
      ]
    }
    margin={
      Array [
        10,
        10,
      ]
    }
    maxRows={Infinity}
    onDrag={[Function]}
    onDragStart={[Function]}
    onDragStop={[Function]}
    onLayoutChange={[Function]}
    onResize={[Function]}
    onResizeStart={[Function]}
    onResizeStop={[Function]}
    preventCollision={false}
    rowHeight={200}
    style={Object {}}
    useCSSTransforms={true}
    verticalCompact={true}
    width={900}
  >
    <div
      className="viz"
      key="Temp Line"
      onClick={[Function]}
    >
      <Viz
        data={
          Array [
            Object {
              "date": "12/12",
              "temp": 2,
            },
            Object {
              "date": "13/12",
              "temp": 0,
            },
            Object {
              "date": "14/12",
              "temp": 5,
            },
            Object {
              "date": "15/12",
              "temp": -3,
            },
          ]
        }
        deviceName="LondonTemp"
        model="VictoryLine"
        name="Temp Line"
        x="date"
        y="temp"
      />
    </div>
    <div
      className="viz"
      key="Humidity Bar"
      onClick={[Function]}
    >
      <Viz
        data={
          Array [
            Object {
              "date": "12/12",
              "hum": 95,
            },
            Object {
              "date": "13/12",
              "hum": 93,
            },
            Object {
              "date": "14/12",
              "hum": 98,
            },
            Object {
              "date": "15/12",
              "hum": 85,
            },
          ]
        }
        deviceName="LondonHumidity"
        model="VictoryBar"
        name="Humidity Bar"
        x="date"
        y="hum"
      />
    </div>
  </ReactGridLayout>
</div>
`;
